gradle.lockfile

```diff
@@ -0,0 +1,239 @@
+com.google.code.gson:gson:2.8.6=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.errorprone:error_prone_annotations:2.3.4=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.guava:failureaccess:1.0.1=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.guava:guava:29.0-jre=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.http-client:google-http-client-jackson2:1.35.0=productionRuntimeClasspath,runtimeClasspath,testRuntimeClasspath
+com.google.http-client:google-http-client-jackson2:1.36.0=compileClasspath
+com.google.http-client:google-http-client-jackson2:1.37.0=testCompileClasspath
+com.google.http-client:google-http-client:1.35.0=productionRuntimeClasspath,runtimeClasspath,testRuntimeClasspath
+com.google.http-client:google-http-client:1.36.0=compileClasspath
+com.google.http-client:google-http-client:1.37.0=testCompileClasspath
+com.google.j2objc:j2objc-annotations:1.3=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.google.protobuf:protobuf-java-util:3.11.4=productionRuntimeClasspath,runtimeClasspath,testRuntimeClasspath
+com.google.protobuf:protobuf-java-util:3.12.2=compileClasspath,testCompileClasspath
+com.google.protobuf:protobuf-java:3.11.4=productionRuntimeClasspath,runtimeClasspath,testRuntimeClasspath
+com.google.protobuf:protobuf-java:3.12.2=compileClasspath,testCompileClasspath
+com.jayway.jsonpath:json-path:2.4.0=testCompileClasspath,testRuntimeClasspath
+com.squareup.okhttp3:logging-interceptor:3.14.9=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.squareup.okhttp3:okhttp:3.14.9=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.squareup.okio:okio:1.17.2=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+com.vaadin.external.google:android-json:0.0.20131108.vaadin1=testCompileClasspath,testRuntimeClasspath
+com.zaxxer:HikariCP:3.4.5=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+commons-codec:commons-codec:1.14=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+io.github.classgraph:classgraph:4.8.69=compileClasspath,productionRuntimeClasspath,runtimeClasspath,testCompileClasspath,testRuntimeClasspath
+io.grpc:grpc-alts:1.29.0=productionRuntimeClasspath,runtimeClasspath,testRuntimeClasspath
+io.grpc:grpc-alts:1.30.0=compileClasspath,testCompileClasspath
```

Comment:
what does it mean to have diffrent versions for `runtime` and `compile`? Should we be worried?

Reply:
we shouldn't be worried, these are selected by the versions specified by the transitive deps that require them, which are different in the two configs.

the (simplified) inheritence:
```
$ gw dependencyInsight --dependency=grpc-alts

> Task :dependencyInsight

io.grpc:grpc-alts:{strictly 1.30.0} -> 1.30.0
\--- compileClasspath

io.grpc:grpc-alts:1.30.0
\--- com.google.api:gax-grpc:1.57.1
     +--- compileClasspath (requested com.google.api:gax-grpc:{strictly 1.57.1})
     \--- com.google.cloud:google-cloud-core-grpc:1.93.7
          +--- compileClasspath (requested com.google.cloud:google-cloud-core-grpc:{strictly 1.93.7})
          \--- org.springframework.cloud:spring-cloud-gcp-starter-trace:1.2.5.RELEASE
               \--- compileClasspath
```

vs

```
$ gw dependencyInsight --dependency=grpc-alts --configuration=runtimeClasspath

> Task :dependencyInsight

io.grpc:grpc-alts:{strictly 1.29.0} -> 1.29.0
\--- runtimeClasspath

io.grpc:grpc-alts:1.29.0
\--- com.google.cloud:libraries-bom:5.3.0
     +--- runtimeClasspath (requested com.google.cloud:libraries-bom:{strictly 5.3.0})
     \--- bio.terra:stairway:0.0.20
          \--- runtimeClasspath

io.grpc:grpc-alts:1.28.1 -> 1.29.0
\--- com.google.api:gax-grpc:1.56.0
     +--- runtimeClasspath (requested com.google.api:gax-grpc:{strictly 1.56.0})
     +--- com.google.cloud:libraries-bom:5.3.0
     |    +--- runtimeClasspath (requested com.google.cloud:libraries-bom:{strictly 5.3.0})
     |    \--- bio.terra:stairway:0.0.20
     |         \--- runtimeClasspath
     +--- com.google.cloud:google-cloud-pubsub:1.105.0
     |    +--- runtimeClasspath (requested com.google.cloud:google-cloud-pubsub:{strictly 1.105.0})
     |    +--- bio.terra:stairway:0.0.20 (requested com.google.cloud:google-cloud-pubsub:1.104.1) (*)
     |    \--- com.google.cloud:libraries-bom:5.3.0 (*)
     \--- com.google.cloud:google-cloud-core-grpc:1.93.4
          +--- runtimeClasspath (requested com.google.cloud:google-cloud-core-grpc:{strictly 1.93.4})
          +--- com.google.cloud:libraries-bom:5.3.0 (*)
          \--- org.springframework.cloud:spring-cloud-gcp-starter-trace:1.2.5.RELEASE (requested com.google.cloud:google-cloud-core-grpc:1.93.7)
               \--- runtimeClasspath
```